pilot/pkg/model/config.go

```diff
@@ -260,9 +260,9 @@ type IstioConfigStore interface {
 	// destination instances.
 	RouteRulesByDestination(destination []*ServiceInstance, domain string) []Config
 
-	// Policy returns a policy for a service version that match at least one of
+	// DestinationPolicy returns a policy for a service version that match at least one of
```

Comment:
If you are renaming/cleaning up - what is a 'policy' ? I never understood the use of the word in this context. Also add few comments on where it is used - that may help a bit. 
AFAIK this returns the configuration used for connecting to a cluster - based on destination rule ( 
is 'rule' and 'policy' the same thing ? Can we call it DestinationRule ).

This seems to be specific to alpha1 - why aren't we renaming to use new names ? 
And to keep alpha1 stable, can't we add a new cleaner method, that returns the alpha3 proto ?

Reply:
The answer to most of it is yes. We already have a separate DestinationRule function.. This is to simply remove confusion on that Policy thing, which has been around for a year!.. There are other uses of policy throughout the code.. 

Also, this interface will be going away in a month. The main goal of this PR is the set of changes to mixer functions to use more understandable names like buildMixerConfigForExternalServices.. instead of a generic BuildMixerConfig..